File: scripts/upload-images.py
import os, glob, time, urllib.request, json

PROJECT_REF = "ogaisocuxygcembgodna"
SERVICE_KEY = open(r"C:\Users\jorda\Downloads\ganjavores-dc-v7\.env.local").read().strip().split('\n')[2].split('=')[1]
BASE_URL = f"https://{PROJECT_REF}.supabase.co/storage/v1/object/public/product-images"

public_dir = r"C:\Users\jorda\Downloads\ganjavores-dc-v7\public\products"
categories = ['flower', 'vapes', 'edibles', 'pre-rolls']

uploaded = 0
failed = 0

for cat in categories:
    folder = os.path.join(public_dir, cat)
    if not os.path.exists(folder):
        continue
    for f in os.listdir(folder):
        if not f.endswith('.png'):
            continue
        
        fpath = os.path.join(folder, f)
        remote_path = f"{cat}/{f}"
        
        try:
            with open(fpath, 'rb') as file_data:
                data = file_data.read()
            
            # Upload to Supabase Storage via REST API
            url = f"{BASE_URL}/{remote_path}?upsert=true"
            req = urllib.request.Request(
                url,
                data=data,
                headers={
                    'Authorization': f'Bearer {SERVICE_KEY}',
                    'apikey': SERVICE_KEY,
                    'Content-Type': 'image/png',
                    'Content-Length': str(len(data))
                },
                method='POST'
            )
            resp = urllib.request.urlopen(req, timeout=15)
            uploaded += 1
        except Exception as e:
            failed += 1
        
        time.sleep(0.05)

print(f"Uploaded: {uploaded}, Failed: {failed}")

File: scripts/upload-images.mjs
import { createClient } from '@supabase/supabase-js'
import { readdirSync, readFileSync } from 'fs'
import { join } from 'path'

const supabase = createClient(
  'https://ogaisocuxygcembgodna.supabase.co',
  'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6Im9nYWlzb2N1eXFnZW1iZ29kbmEiLCJhdWQiOiJzdXBhYmFzZS1yZWxlYXNlcmQiLCJpYXQiOjE3NTgyNjk4MDB9.Mo1Y'
)
const bucket = supabase.storage.from('product-images')
const publicDir = './public/products'
const categories = ['flower', 'vapes', 'edibles', 'pre-rolls']

async function upload() {
  let uploaded = 0
  let failed = 0

  for (const cat of categories) {
    try {
      const files = readdirSync(join(publicDir, cat))
      for (const file of files) {
        if (!file.endsWith('.png')) continue
        try {
          const buffer = readFileSync(join(publicDir, cat, file))
          const { error } = await bucket.upload(cat + '/' + file, buffer, { upsert: true })
          if (!error) {
            uploaded++
          } else {
            failed++
          }
        } catch {
          failed++
        }
      }
    } catch {}
  }
  console.log('Uploaded: ' + uploaded + ', Failed: ' + failed)
}
upload()
